network/ovssnat/ovssnat.go

```diff
@@ -0,0 +1,411 @@
+	log.Printf("[ovs] Setting link %v netns %v.", client.containerSnatVethName, netnsPath)
+	return netlink.SetLinkNetNs(client.containerSnatVethName, nsID)
+}
+
+func (client *OVSSnatClient) SetupSnatContainerInterface() error {
+	if err := epcommon.SetupContainerInterface(client.containerSnatVethName, azureSnatIfName); err != nil {
+		return err
+	}
+
+	client.containerSnatVethName = azureSnatIfName
+
+	return nil
+}
+
+func (client *OVSSnatClient) AllowInboundFromHostToNC() error {
+	bridgeIP, _, _ := net.ParseCIDR(client.snatBridgeIP)
+	containerIP, _, _ := net.ParseCIDR(client.localIP)
+
+	if err := iptables.CreateCNIChain(iptables.Filter, iptables.CNIOutputChain); err != nil {
+		log.Printf("AllowInboundFromHostToNC: Creating %v failed with error: %v", iptables.CNIOutputChain, err)
+		return err
+	}
+
+	if err := iptables.InsertIptableRule(iptables.Filter, iptables.Output, "", iptables.CNIOutputChain); err != nil {
+		log.Printf("AllowInboundFromHostToNC: Inserting forward rule to %v failed with error: %v", iptables.CNIOutputChain, err)
+		return err
+	}
+
+	// Allow Host to NC traffic
+	matchCondition := fmt.Sprintf("-s %s -d %s", bridgeIP.String(), containerIP.String())
+	err := iptables.InsertIptableRule(iptables.Filter, iptables.CNIOutputChain, matchCondition, iptables.Accept)
+	if err != nil {
+		log.Printf("AllowInboundFromHostToNC: Inserting output rule failed: %v", err)
+		return err
+	}
+
+	// Create cniinput chain
+	if err := iptables.CreateCNIChain(iptables.Filter, iptables.CNIInputChain); err != nil {
+		log.Printf("AllowInboundFromHostToNC: Creating %v failed with error: %v", iptables.CNIInputChain, err)
+		return err
+	}
+
+	// Forward from Input to cniinput chain
+	if err := iptables.InsertIptableRule(iptables.Filter, iptables.Input, "", iptables.CNIInputChain); err != nil {
+		log.Printf("AllowInboundFromHostToNC: Inserting forward rule to %v failed with error: %v", iptables.CNIInputChain, err)
+		return err
+	}
+
+	matchCondition = fmt.Sprintf(" -i %s -m state --state ESTABLISHED,RELATED", SnatBridgeName)
+	err = iptables.InsertIptableRule(iptables.Filter, iptables.CNIInputChain, matchCondition, iptables.Accept)
+	if err != nil {
+		log.Printf("AllowInboundFromHostToNC: Inserting input rule failed: %v", err)
+		return err
+	}
+
+	snatContainerVeth, _ := net.InterfaceByName(client.containerSnatVethName)
+
+	log.Printf("Adding static arp entry for ip %s mac %s", containerIP, snatContainerVeth.HardwareAddr.String())
+	err = netlink.AddOrRemoveStaticArp(netlink.ADD, SnatBridgeName, containerIP, snatContainerVeth.HardwareAddr)
+	if err != nil {
+		log.Printf("AllowInboundFromHostToNC: Error adding static arp entry for ip %s mac %s: %v", containerIP, snatContainerVeth.HardwareAddr.String(), err)
+	}
+
+	return err
+}
+
+func (client *OVSSnatClient) DeleteInboundFromHostToNC() error {
+	bridgeIP, _, _ := net.ParseCIDR(client.snatBridgeIP)
```

Comment:
we can create a func to return these 2 variables, seems like we write this out several times